cstore_compression.c

```diff
@@ -0,0 +1,165 @@
+/*-------------------------------------------------------------------------
+ *
+ * cstore_compression.c
+ *
+ * This file contains compression/decompression functions definitions
+ * used in cstore_fdw.
+ *
+ * Copyright (c) 2016, Citus Data, Inc.
+ *
+ * $Id$
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+#include "cstore_fdw.h"
+
+#if PG_VERSION_NUM >= 90500
+#include "common/pg_lzcompress.h"
+#else
+#include "utils/pg_lzcompress.h"
+#endif
+
+
+
+
+#if PG_VERSION_NUM >= 90500
+/*
+ *	The information at the start of the compressed data. This decription is taken
+ *	from pg_lzcompress in pre-9.5 version of PostgreSQL.
+ */
+typedef struct CStoreCompressHeader
```

Comment:
Can't we move this struct definition, the following `#define`s and `CompressBuffer` & `DecompressBuffer` declartions to `cstore_compression.h` ? I think I'd be more compact for compression related stuff.

Also, we generally use `define` macros in the headers and having the declarations in `cstrore_fdw.h` doesn't look the best approach.